cpp/src/join/distinct_hash_join.cu

```diff
@@ -102,6 +110,9 @@ distinct_hash_join::distinct_hash_join(cudf::table_view const& build,
 {
   CUDF_FUNC_RANGE();
   CUDF_EXPECTS(0 != this->_build.num_columns(), "Hash join build table is empty");
+  CUDF_EXPECTS(load_factor > 0 && load_factor <= 1,
```

Comment:
Unlikely scenario but should we use an `epsilon` for double comparisons here to handle edge cases?

Reply:
Ha, good question. Epsilon is typically only needed when checking for equality, not for lexicographical comparisons. In this case, it might be relevant if you're comparing the input occupancy to 1.0 exactly. However, in practice, we never aim for 100% occupancy due to performance concerns. So in general, using epsilon here isn’t necessary.

Reply:
Agreed we don't aim for 1 but I'd say we should still handle the edge case or change the condition to `0 < load_factor < 1`